x-pack/plugins/rule_registry/server/rule_data_plugin_service/rule_data_plugin_service.test.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { loggerMock } from '@kbn/logging/mocks';
+import { RuleDataService } from './rule_data_plugin_service';
+import { elasticsearchServiceMock } from 'src/core/server/mocks';
+import { AlertConsumers } from '@kbn/rule-data-utils/alerts_as_data_rbac';
+import { Dataset } from './index_options';
+import { RuleDataClient } from '../rule_data_client/rule_data_client';
+import { createRuleDataClientMock as mockCreateRuleDataClient } from '../rule_data_client/rule_data_client.mock';
+
+jest.mock('../rule_data_client/rule_data_client', () => ({
```

Comment:
Usually this means we don't use any dependency injection model :)
Apart from this, the tests look good :)

Reply:
@ersin-erdal Exactly. I had troubles mocking constructors like RuleDataClient or ResourceInstaller in the rule_data_plugin_service.test. I was thinking to refactor using dependency injection instead of directly importing the dependencies. Let's discuss about this during our today's meeting